### ISSUE TITLE
In-app messages cause android app crash "Software rendering doesn't support hardware bitmaps"

### ISSUE THREAD

**Issue by kauz-mobilepay** [mixpanel/mixpanel-android#711]
Hi,

Mixpanel in-app messages are causing "Software rendering doesn't support hardware bitmaps" crashes when capturing screens that contain hardware accelerated images: 
```
com.mixpanel.android.util.ActivityImageUtils.getScaledScreenshot (ActivityImageUtils.java:15)
com.mixpanel.android.util.ActivityImageUtils.getHighlightColorFromBackground (ActivityImageUtils.java:43)
com.mixpanel.android.mpmetrics.MixpanelAPI$PeopleImpl$2.run (MixpanelAPI.java:2509)
android.app.Activity.runOnUiThread (Activity.java:6922)
com.mixpanel.android.mpmetrics.MixpanelAPI$PeopleImpl.showGivenOrAvailableNotification (MixpanelAPI.java:2482)

```

PixelCopy could be used for solving this.

**Comment by clhols** [comment]
We are still seeing this crash our app in Mixpanel 5.8.5.

**Comment by zihejia** [comment]
hi @kauz-mobilepay @clhols , we have released v5.8.7 which has a fix for this issue. I'm closing this one now, please feel free to reopen it if the problem still exists.

**Comment by gosr** [comment]
Hi @zihejia, this crash is also occurring in v5.9.1:

> Fatal Exception: java.lang.IllegalArgumentException: Software rendering doesn't support hardware bitmaps
       at android.graphics.BaseCanvas.onHwBitmapInSwMode(BaseCanvas.java:673)
       at android.graphics.BaseCanvas.throwIfHwBitmapInSwMode(BaseCanvas.java:680)
       at android.graphics.BaseCanvas.throwIfCannotDraw(BaseCanvas.java:84)
       at android.graphics.BaseCanvas.drawBitmap(BaseCanvas.java:142)
       at android.graphics.Canvas.drawBitmap(Canvas.java:1648)
       at android.graphics.drawable.BitmapDrawable.draw(BitmapDrawable.java:548)
       at android.widget.ImageView.onDraw(ImageView.java:1453)
       at android.view.View.draw(View.java:23901)
       at android.view.View.draw(View.java:23773)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at androidx.constraintlayout.widget.ConstraintLayout.dispatchDraw(ConstraintLayout.java:1975)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23904)
       at android.view.View.draw(View.java:23773)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at androidx.recyclerview.widget.RecyclerView.drawChild(RecyclerView.java:5030)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23904)
       at androidx.recyclerview.widget.RecyclerView.draw(RecyclerView.java:4429)
       at android.view.View.draw(View.java:23773)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at androidx.constraintlayout.widget.ConstraintLayout.dispatchDraw(ConstraintLayout.java:1975)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at androidx.fragment.app.FragmentContainerView.drawChild(FragmentContainerView.java:261)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at androidx.fragment.app.FragmentContainerView.dispatchDraw(FragmentContainerView.java:249)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at androidx.constraintlayout.widget.ConstraintLayout.dispatchDraw(ConstraintLayout.java:1975)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at androidx.coordinatorlayout.widget.CoordinatorLayout.drawChild(CoordinatorLayout.java:1277)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23771)
       at android.view.ViewGroup.drawChild(ViewGroup.java:5336)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:5093)
       at android.view.View.draw(View.java:23904)
       at com.android.internal.policy.DecorView.draw(DecorView.java:1282)
       at com.mixpanel.android.util.ActivityImageUtils.getScaledScreenshot(ActivityImageUtils.java:31)
       at com.mixpanel.android.util.ActivityImageUtils.getHighlightColorFromBackground(ActivityImageUtils.java:52)
       at com.mixpanel.android.mpmetrics.MixpanelAPI$PeopleImpl$2.run(MixpanelAPI.java:2541)
       at android.app.Activity.runOnUiThread(Activity.java:7154)
       at com.mixpanel.android.mpmetrics.MixpanelAPI$PeopleImpl.showGivenOrAvailableNotification(MixpanelAPI.java:2514)
       at com.mixpanel.android.mpmetrics.MixpanelAPI$PeopleImpl.showGivenNotification(MixpanelAPI.java:2333)
       at com.mixpanel.android.mpmetrics.MixpanelAPI.track(MixpanelAPI.java:2928)
       at com.mixpanel.android.mpmetrics.MixpanelAPI.track(MixpanelAPI.java:900)
       at com.mixpanel.android.mpmetrics.MixpanelAPI.trackMap(MixpanelAPI.java:842)
